GPU/Directx9/StateMappingDX9.cpp

```diff
@@ -31,41 +31,37 @@
 
 namespace DX9 {
 
-static const D3DBLEND aLookup[11] = {
-	D3DBLEND_DESTCOLOR,
-	D3DBLEND_INVDESTCOLOR,
-	D3DBLEND_SRCALPHA,
-	D3DBLEND_INVSRCALPHA,
-	D3DBLEND_DESTALPHA,
-	D3DBLEND_INVDESTALPHA,
-	D3DBLEND_SRCALPHA,	// should be 2x
-	D3DBLEND_INVSRCALPHA,	 // should be 2x
-	D3DBLEND_DESTALPHA,	 // should be 2x
-	D3DBLEND_INVDESTALPHA,	 // should be 2x	-	and COLOR?
-	D3DBLEND_BLENDFACTOR,	// FIXA
-};
-
-static const D3DBLEND bLookup[11] = {
+static const D3DBLEND dxBlendFactorLookup[(size_t)BlendFactor::COUNT] = {
+	D3DBLEND_ZERO,
+	D3DBLEND_ONE,
 	D3DBLEND_SRCCOLOR,
 	D3DBLEND_INVSRCCOLOR,
+	D3DBLEND_DESTCOLOR,
+	D3DBLEND_INVDESTCOLOR,
 	D3DBLEND_SRCALPHA,
 	D3DBLEND_INVSRCALPHA,
 	D3DBLEND_DESTALPHA,
 	D3DBLEND_INVDESTALPHA,
-	D3DBLEND_SRCALPHA,	// should be 2x
-	D3DBLEND_INVSRCALPHA,	 // should be 2x
-	D3DBLEND_DESTALPHA,	 // should be 2x
-	D3DBLEND_INVDESTALPHA,	 // should be 2x
-	D3DBLEND_BLENDFACTOR,	// FIXB
+	D3DBLEND_BLENDFACTOR,
+	D3DBLEND_INVBLENDFACTOR,
+	D3DBLEND_BLENDFACTOR,
+	D3DBLEND_INVBLENDFACTOR,
+#if 0   // TODO: Requires D3D9Ex
```

Comment:
But doesn't work (maybe I'm doing it wrong):
https://github.com/hrydgard/ppsspp/compare/master...unknownbrackets:d3d9-dualsrc

-[Unknown]